openhcl/virt_mshv_vtl/src/processor/hardware_cvm/mod.rs

```diff
@@ -350,11 +362,57 @@ impl<T: CpuIo, B: HardwareIsolatedBacking> UhHypercallHandler<'_, '_, T, B> {
         self.vp.switch_vtl(self.intercepted_vtl, GuestVtl::Vtl1);
         self.vp.backing.cvm_state_mut().exit_vtl = GuestVtl::Vtl1;
 
+        // TODO GUEST VSM: reevaluate if the return reason should be set here or
```

Comment:
FYI @smalis-msft I think you are going to want to move this to the exit handling code. Figured I'll just let you decide if/how you want to pass any information over.